### ISSUE TITLE
Parallel gateway check

### ISSUE THREAD

**Issue by johnuopini** [moul/assh#351]
Right now ASSH will try to connect using gateways in order and then upon timeout will go to the next. Wouldn't be much better to have a set of threads spawn for every gateway so all can be tested in parallel and close the connections of the pending ones as soon as the first becomes available?

This is currently one of the biggest issues i have since i switch environment quite often and i have 4 different ways to access my first jump box at work. Currently i use a python sshccommand wrapper that does this in place of assh but would be nice to have everything in just one config. 

If this is not for all maybe this could be added as an option.

**Comment by Lilvelli** [comment]
+923049481565
